### PR TITLE
Refactor: Added Navigation Helper to generate navbar items dynamically

### PR DESCRIPTION
**Pull request by keshavbiswa**
## Summary
We were using a lot of static anchor tags to generate our navbar items which were quite redundant.

## Type of change
- Added Navigation Helper
- Rendered Navbars using `nav_helper`
Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### Checklist:

- [ ] I have manually tested all workflows
- [ ] I have performed a self-review of my own code
- [ ] I have added automated tests for my code

### CODE REVIEW

**Review by rohitjoshixyz — APPROVED** [review]
LGTM

File: app/helpers/navigation_helper.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        {
+          url: time_tracking_index_path,
+          title: I18n.t("navbar.time_tracking"),
+          permitted: Pundit.policy!(current_user, :timesheet_entry).index?,
+          data: { cy: "time-tracking-tab" }
+        },
+        {
+          url: team_index_path,
+          title: I18n.t("navbar.team"),
+          permitted: Pundit.policy!(current_user, :timesheet_entry).index?,
```
Comment (rohitjoshixyz) [inline app/helpers/navigation_helper.rb:30]: Check policy class

**Review by rohitjoshixyz — CHANGES_REQUESTED** [review]
Added a comment

**Comment by supriya3105** [comment]
@keshavbiswa Please check the review comments.

**Comment by keshavbiswa** [comment]
@rohitjoshixyz @supriya3105 Addressed the code review concerns. Thanks for the catch @rohitjoshixyz

**Comment by supriya3105** [comment]
@keshavbiswa The CI checks are failing on this PR. Please check

**Pull request merged**